src/pred_set/src/more_theories/finite_setScript.sml

```diff
@@ -1168,4 +1191,61 @@ Proof
   rw[fIN_def]
 QED
 
+Theorem fset_ABS_MAP:
+  fset_ABS (MAP f l) = fIMAGE f (fset_ABS l)
+Proof
+  rw[EXTENSION, fIN_IN, toSet_fset_ABS, MEM_MAP]
+  \\ metis_tac[]
+QED
+
+Theorem fset_REP_fEMPTY[simp]:
+  fset_REP fEMPTY = []
+Proof
+  rw[rich_listTheory.NIL_NO_MEM, MEM_fset_REP]
+QED
+
+Theorem fIN_fset_ABS:
+  fIN x (fset_ABS l) <=> MEM x l
+Proof
+  rw[fIN_def]
+  \\ mp_tac fset_QUOTIENT
+  \\ rewrite_tac[quotientTheory.QUOTIENT_def,fsequiv_def]
+  \\ rpt strip_tac
+  \\ `set (fset_REP (fset_ABS l)) = set l`
+  suffices_by (
+    rewrite_tac[pred_setTheory.EXTENSION]
+    \\ metis_tac[] )
+  \\ asm_simp_tac std_ss []
+QED
+
+Theorem fBIGUNION_fset_ABS_FOLDL_aux[local]:
+  !l s. FOLDL fUNION s l = fUNION s (fBIGUNION (fset_ABS l))
+Proof
+  Induct \\ rw[fBIGUNION_def, GSYM fEMPTY_def]
+  \\ rw[GSYM fUNION_ASSOC] \\ AP_TERM_TAC
+  \\ rw[EXTENSION]
+  \\ qmatch_goalsub_abbrev_tac`_ \/ fIN e s <=> fIN e hs`
+  \\ `hs = fUNION h s`
+  by (
+    rw[Once (GSYM toSet_11)]
+    \\ rw[Abbr`hs`, Abbr`s`, toSet_fset_ABS]
+    \\ rw[pred_setTheory.EXTENSION, MEM_FLAT, PULL_EXISTS, MEM_MAP]
+    \\ rw[MEM_fset_REP, fIN_fset_ABS, GSYM fIN_IN]
+    \\ metis_tac[] )
+  \\ rw[]
+QED
+
+Theorem fBIGUNION_fset_ABS_FOLDL:
+  fBIGUNION (fset_ABS l) = FOLDL fUNION fEMPTY l
+Proof
+  rw[fBIGUNION_fset_ABS_FOLDL_aux]
+QED
+
+Theorem fIMAGE_MAP:
```

Comment:
One of this and `fset_ABS_MAP` must be redundant.

Reply:
Removed that one.